app/helpers/device_stories_helper.rb

```diff
@@ -74,43 +74,38 @@ def last_air_quality_values(last_values)
   end
 
   # The names should correspond to the names in the query
-  def sensor_names
-    { 'radiation_sensors' => %w(lnd_7128ec lnd_7318c lnd_712u lnd_7318u lnd_78017w lnd7318u lnd7128c),
-      'air_sensors' => %w(pms_pm10_0 pms_pm02_5 pms_pm01_0),
-      'bat_voltage' => ['bat_voltage'],
-      'temperature_C' => ['temperature_C'],
-      'temperature_F' => ['temperature_F'],
-      'humidity' => ['humidity'],
-      'pressure' => ['pressure'],
-      'charging' => ['charging'] }
-  end
+  CATEGORY_SENSOR_NAMES = {
+    'radiation_sensors' => %w(lnd_7128ec lnd_7318c lnd_712u lnd_7318u lnd_78017w lnd7318u lnd7128c),
+    'air_sensors' => %w(pms_pm10_0 pms_pm02_5 pms_pm01_0),
+    'bat_voltage' => ['bat_voltage'],
+    'temperature_C' => ['temperature_C'],
+    'temperature_F' => ['temperature_F'],
+    'humidity' => ['humidity'],
+    'pressure' => ['pressure'],
+    'charging' => ['charging']
+  }.freeze
 
-  def get_sensor_data() # rubocop:disable all
-    q = IngestMeasurement.query_sensor_data(@device_story.device_urn)
-    all_hashes = {}
-    sensor_names.values.each do |sensor_type|
-      sensor_type_hashes = []
-      sensor_type.each do |sensor|
-        hash_sensor = {}
-        sensor_exists = false
-        q.response['aggregations']['sensor_data']['buckets'].each do |bucket|
-          date = Time.at(bucket['key'] / 1000.0).strftime('%Y-%m-%d %H')
-          sensor_value = bucket[sensor]['value']
-          if sensor_value then sensor_exists = true end
-          hash_sensor.merge!(date => sensor_value)
-        end
-        if sensor_exists then sensor_type_hashes.push({ "name": sensor, "data": hash_sensor }) end
+  def sensor_data_by_sensors(sensor_data)
+    data_by_dates = convert_by_dates(sensor_data.dig('aggregations', 'sensor_data', 'buckets') || [])
```

Comment:
I put this on dev but it's getting `undefined method `dig' for #<Elasticsearch::Model::Response::Response:0x0000000007d4ee58>`

https://5bc3d4f7330a4459881197a85659caf5.us-west-2.aws.found.io:9243/app/apm#/services/safecastapi-dev/errors/1baaae338359071560c867ba08f88b5c?rangeFrom=now-15m&rangeTo=now&refreshPaused=true&refreshInterval=0